src/models/visualisation.py

```diff
@@ -0,0 +1,213 @@
+        )
+
+        choropleth = folium.Choropleth(
+            polygon_gdf,
+            data=polygon_gdf,
+            key_on="feature.properties.index",
+            columns=["index", "color_index"],
+            fill_color="YlOrBr",
+            name=name + "_polygons",
+        )
+        choropleth.add_to(m)
+
+        # adding popup markers with class name
+        folium.features.GeoJsonPopup(fields=[color_column], labels=True).add_to(
+            choropleth.geojson
+        )
+
+    # Adding graph data
+    if graph is not None:
+        node_gdf, edge_gdf = create_node_edge_geometries(graph, crs=crs)
+
+        # add graph edges to map
+        if not edge_gdf.empty:
+            edges = folium.features.GeoJson(
+                edge_gdf,
+                name=name + "_graph_edges",
+                style_function=get_style_function("#dd0000"),
+            )
+            edges.add_to(m)
+
+        # add graph nodes/vertices to map
+        node_marker = folium.vector_layers.Circle(radius=100, color="black")
+        nodes = folium.features.GeoJson(
+            node_gdf, marker=node_marker, name=name + "_graph_vetrices"
+        )
+        nodes.add_to(m)
+
+    if add_layer_control:
+        folium.LayerControl().add_to(m)
+
+    return m
+
+
+def create_node_edge_geometries(
+    G: nx.Graph, crs: str = UTM35N
+) -> Tuple[gpd.GeoDataFrame, gpd.GeoDataFrame]:
+    """
+    This function creates node and edge geometries for the networkx graph G and returns
+    them in two GeoDataFrames. The output can be used for plotting a graph.
+
+    Args:
+        G (nx.Graph): graph with nodes and edges
+        crs (str, optional): coordinate reference system. Defaults to UTM35N.
+
+    Returns:
+        Tuple[gpd.GeoDataFrame, gpd.GeoDataFrame]: dataframes of nodes and edges
+            respectively.
+    """
+
+    node_gdf = gpd.GeoDataFrame(columns=["id", "geometry"])
+    rep_points = G.nodes(data="representative_point")
+    for idx, rep_point in rep_points:
+        node_gdf.loc[idx] = [idx, rep_point]
+
+    edge_gdf = gpd.GeoDataFrame(columns=["id", "geometry"])
+    for idx, (node_a, node_b) in enumerate(G.edges()):
+        point_a = rep_points[node_a]
+        point_b = rep_points[node_b]
+        line = shapely.geometry.LineString([point_a, point_b])
+
+        edge_gdf.loc[idx] = [idx, line]
+
+    node_gdf = node_gdf.set_crs(crs)
+    edge_gdf = edge_gdf.set_crs(crs)
+
+    return node_gdf, edge_gdf
+
+
+def get_style_function(color: str = "#ff0000") -> Callable[[], dict]:
+    """
+    This function returns a lambda function the returns a dict with the `color` given.
+    The returned lambda function can be used as a style function for folium.
+
+    Args:
+        color (str, optional): color to be used in dict. Defaults to "#ff0000".
+
+    Returns:
+        Callable[[], dict]: style function
+    """
+
+    return lambda x: {"fillColor": color, "color": color}
+
+
+def add_cez_to_map(
+    m: folium.Map,
```

Comment:
nit: variable naming
```suggestion
    map: folium.Map,
```